src/Diagnostics.DataProviders/LogAnalyticsClientBase.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Diagnostics.DataProviders.Interfaces;
+using Diagnostics.Logger;
+using Microsoft.Azure.OperationalInsights;
+using Microsoft.Azure.OperationalInsights.Models;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace Diagnostics.DataProviders
+{
+    public abstract class LogAnalyticsClientBase : ILogAnalyticsClient
+    {
+        public abstract OperationalInsightsDataClient client { get; set; }
+        public abstract string _requestId { get; set; }
+
+        public async Task<DataTable> ExecuteQueryAsync(string query)
+        {
+            var timeTakenStopWatch = new Stopwatch();
+            QueryResults queryResults;
+            Exception dataProviderException = null;
+
+            try
+            {
+                timeTakenStopWatch.Start();
+                queryResults = await client.QueryAsync(query);
+            }
+            catch (Exception ex)
+            {
+                dataProviderException = ex;
+                throw;
+            }
+            finally
+            {
+                timeTakenStopWatch.Stop();
+                var StopTime = DateTime.Now;
+                var StartTime = StopTime.AddMilliseconds(-timeTakenStopWatch.ElapsedMilliseconds);
+
+                if (dataProviderException != null)
+                {
+                    DiagnosticsETWProvider.Instance.LogDataProviderException(_requestId,
+                        "K8SELogAnalyticsDataProvider", StartTime.ToString("HH:mm:ss.fff"),
```

Comment:
move the data provider name - "K8SELogAnalyticsDataProvider" as well to the derived class.